src/setup-elixir.js

```diff
@@ -1,97 +1,356 @@
 const core = require('@actions/core')
 const {exec} = require('@actions/exec')
-const {installElixir, installOTP} = require('./installer')
+const installer = require('./installer')
 const path = require('path')
 const semver = require('semver')
 const https = require('https')
 const {
-  fstat,
   promises: {readFile},
 } = require('fs')
 
-main().catch(err => {
+try {
+  main().catch((err) => {
+    core.setFailed(err.message)
+  })
+} catch (err) {
   core.setFailed(err.message)
-})
+}
 
 async function main() {
-  checkPlatform()
+  const maybePlatformIsOKError = installer.checkPlatform()
+  if (maybePlatformIsOKError !== 'ok') {
+    throw new Error(maybePlatformIsOKError)
+  }
 
   const osVersion = getRunnerOSVersion()
+  const otpBuildsListing = {
+    remote: `https://repo.hex.pm/builds/otp/${osVersion}/builds.txt`,
+  }
```

Comment:
This seems like unnecessary complexity, can we pass the strings directly without wrapping in an object?

Reply:
You can, if you decide that `builds.txt` is always remote. Otherwise, I have it there for testing (which uses a string, not a URL). I could also go for "parsing as URL", but decided this would be more explicit. What do you prefer?

Reply:
f79efb2